### PR TITLE
fix: delete the correct dashboard cache key

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

This PR fixes a bug from #11234 where cache keys for dashboard bootstrap data are incorrectly deleted.

We noticed this when some dashboard failed to update cache when updating charts. 

`delete_memoized` from Flask caching treats instance method and class method differently:

```
        Delete memoized is also smart about instance methods vs class methods.
        When passing a instancemethod, it will only clear the cache related
        to that instance of that object. (object uniqueness can be overridden
        by defining the __repr__ method, such as user id).
        When passing a classmethod, it will clear all caches related across
        all instances of that class.
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

1. Create a dashboard and add a chart
2. Change chart title in Dashboard edit mode
3. Explore the chart, make updates and save
4. Return to dashboard and refresh. In base, you may see the chart is not updated. After the fix, the chart should update.

### ADDITIONAL INFORMATION

- [x] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/dashboard.py [hunk 1]
```diff
@@ -294,7 +293,9 @@ def update_thumbnail(self) -> None:
 
     @debounce(0.1)
     def clear_cache(self) -> None:
-        cache.delete_memoized(self.full_data)
+        # pylint: disable=no-member
```
Comment (etr2460) [inline superset/models/dashboard.py:296]: why do we need to disable this lint error here?

File: superset/models/dashboard.py [hunk 2]
```diff
@@ -252,8 +252,7 @@ def data(self) -> Dict[str, Any]:
         }
 
     @cache.memoize(
-        # manually maintain cache key version
-        make_name=lambda fname: f"{fname}-v1",
+        make_name=lambda fname: f"{fname}-v0.1",
```
Comment (etr2460) [inline superset/models/dashboard.py:255]: why not `v2`?

**Review by etr2460 — COMMENTED** [review]
a couple questions

File: superset/models/dashboard.py [hunk 3]
```diff
@@ -294,7 +293,7 @@ def update_thumbnail(self) -> None:
 
     @debounce(0.1)
     def clear_cache(self) -> None:
-        cache.delete_memoized(self.full_data)
```
Comment (ktmud) [inline superset/models/dashboard.py:297]: The original code only hash version related to the function:  https://github.com/sh4nks/flask-caching/blob/v1.9.0/flask_caching/__init__.py#L1051-L1052

File: superset/models/dashboard.py (hunk 2 above)
Comment (etr2460) [reply superset/models/dashboard.py:255]: fair enough

**Review by etr2460 — APPROVED** [review]
thanks for the fix!

**Comment by ktmud** [comment]
Updated the cache key nonetheless!

**Pull request merged**